doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,377 @@
+.. sectnum::
+
+#########################
+Amazon SageMaker Debugger
+#########################
+
+
+Amazon SageMaker Debugger allows you to detect anomalies while training your machine learning model by emitting relevant data during training, storing the data and then analyzing it.
+
+.. contents::
+
+Background
+==========
+
+Amazon SageMaker provides every developer and data scientist with the ability to build, train, and deploy machine learning models quickly. Amazon SageMaker is a fully-managed service that encompasses the entire machine learning workflow. You can label and prepare your data, choose an algorithm, train a model, and then tune and optimize it for deployment. You can deploy your models to production with Amazon SageMaker to make predictions at lower costs than was previously possible.
+
+`SageMaker Debugger <https://docs.aws.amazon.com/sagemaker/latest/dg/train-debugger.html>`__ provides a way to hook into the training process and emit debug artifacts (a.k.a. "tensors") that represent the training state at any given point in the training lifecycle. Debugger then stores the data in real time and provides a way for you to analyze the data through a concept called "rule". Rules encapsulate logic to analyze tensors and react to anomalies.
+
+Setup
+=====
+
+To get started, you must satisfy the following prerequisites:
+
+* Specify an AWS Region where you'll train your model.
+* To give Amazon SageMaker the access to your data in Amazon Simple Storage Service (Amazon S3) needed to train your model, create an IAM role ARN. See the `AWS IAM documentation <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html>`__ for how to fine tune the permissions needed.
+
+Capture real-time debugging data during model training in Amazon SageMaker
+==========================================================================
+
+To enable data emission for debugging model training, Amazon SageMaker initializes a "hook" which attaches itself to the training process and emits data necessary for debugging i.e. tensors. To provide the hook's configuration, specify the new option called ``DebuggerHookConfig`` when training your model. For more about the ``DebuggerHookConfig`` object, see the `API documentation <https://sagemaker.readthedocs.io/en/stable/debugger.html#sagemaker.debugger.DebuggerHookConfig>`__.
```

Comment:
comma before "i.e."

also don't call `DebuggerHookConfig` "new" - it won't be new after some time and nobody will remember to update it 😂

Reply:
Good catch. Thanks. Fixed.